cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -786,6 +878,10 @@ const FunctionDoc max_element_wise_doc{
     {"*args"},
     "ElementWiseAggregateOptions"};
 
+const FunctionDoc between_doc{"Check if values are in a range x <= y <= z",
```

Comment:
I think that `x <=  y <= z` is better for the default because pandas and SQL's `BETWEEN` use it.
But the implementation in this pull request uses `x < y < z`.

Reply:
Can change this to `x <= y <= z`.

Reply:
Added options to allow selection of 
- `x <= y <= z`
- `x < y <= z`
- `x <= y < z`
- `x < y < z`